contrib/TestHarness2/test_harness/config.py

```diff
@@ -0,0 +1,260 @@
+        self.write_run_times_args = {'help': 'Write back probabilities after each test run',
+                                     'action': 'store_true'}
+        self.unseed_check_ratio: float = 0.05
+        self.unseed_check_ratio_args = {'help': 'Probability for doing determinism check'}
+        self.test_dirs: List[str] = ['slow', 'fast', 'restarting', 'rare', 'noSim']
+        self.test_dirs_args: dict = {'nargs': '*', 'help': 'test_directories to look for files in'}
+        self.trace_format: str = 'json'
+        self.trace_format_args = {'choices': ['json', 'xml'], 'help': 'What format fdb should produce'}
+        self.crash_on_error: bool = True
+        self.crash_on_error_args = {'long_name': 'no_crash', 'action': 'store_false',
+                                    'help': 'Don\'t crash on first error'}
+        self.max_warnings: int = 10
+        self.max_warnings_args = {'short_name': 'W'}
+        self.max_errors: int = 10
+        self.max_errors_args = {'short_name': 'E'}
+        self.old_binaries_path: Path = Path('/app/deploy/global_data/oldBinaries/')
+        self.old_binaries_path_args = {'help': 'Path to the directory containing the old fdb binaries'}
+        self.use_valgrind: bool = False
+        self.use_valgrind_args = {'action': 'store_true'}
+        self.buggify = BuggifyOption('random')
+        self.buggify_args = {'short_name': 'b', 'choices': ['on', 'off', 'random']}
+        self.pretty_print: bool = False
+        self.pretty_print_args = {'short_name': 'P', 'action': 'store_true'}
+        self.clean_up: bool = True
+        self.clean_up_args = {'long_name': 'no_clean_up', 'action': 'store_false'}
+        self.run_dir: Path = Path('tmp')
+        self.joshua_seed: int = random.randint(0, 2 ** 32 - 1)
+        self.joshua_seed_args = {'short_name': 's', 'help': 'A random seed', 'env_name': 'JOSHUA_SEED'}
+        self.print_coverage = False
+        self.print_coverage_args = {'action': 'store_true'}
+        self.binary = Path('bin') / ('fdbserver.exe' if os.name == 'nt' else 'fdbserver')
+        self.binary_args = {'help': 'Path to executable'}
+        self.hit_per_runs_ratio: int = 20000
+        self.hit_per_runs_ratio_args = {'help': 'How many test runs should hit each code probe at least once'}
+        self.output_format: str = 'xml'
+        self.output_format_args = {'short_name': 'O', 'choices': ['json', 'xml'],
+                                   'help': 'What format TestHarness should produce'}
+        self.include_test_files: str = r'.*'
+        self.include_test_files_args = {'help': 'Only consider test files whose path match against the given regex'}
+        self.exclude_test_files: str = r'.^'
+        self.exclude_test_files_args = {'help': 'Don\'t consider test files whose path match against the given regex'}
+        self.include_test_names: str = r'.*'
+        self.include_test_names_args = {'help': 'Only consider tests whose names match against the given regex'}
```

Comment:
it uses the name as opposed to the file path. For `include_test_files` you can pass something like `tests/fast/.*` which wouldn't work for `include_test_names`.

The name is defined as `testClass` (if it exists`) or `testName` (if it doesn't). But I guess you have a point that switching the definition (saying test class is `testClass` if present, `testName` otherwise) would be more intuitive. So I am going to change this.